### PR TITLE
[RND-525] - DRAFT - Adds support for ElasticSearch

### PR DESCRIPTION
**Pull request by DavidJGapCR**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] I have signed all of the commits on this PR.
- [ ] I have agreed to the Ed-Fi Individual Contributors' License
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Meadowlark-js/backends/meadowlark-search-backend/src/repository/ClientSearch.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { Client as OpenSearchClient } from '@opensearch-project/opensearch';
+import { Client as ElasticClient } from '@elastic/elasticsearch';
+import { /* Config, */ Logger } from '@edfi/meadowlark-utilities';
+import { BasicAuth } from '@opensearch-project/opensearch/lib/pool';
+
+export type ClientSearch = OpenSearchClient | ElasticClient;
+
+const moduleName = 'opensearch.repository.Db';
+
+function getSearchAPI() {
```
Comment (stephenfuqua) [inline Meadowlark-js/backends/meadowlark-search-backend/src/repository/ClientSearch.ts:10]: What do you think about using `getSearchProviderName` for this function?

File: Meadowlark-js/backends/meadowlark-search-backend/src/repository/ClientSearch.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  // return Config.get<string>('SEARCH_API');
```
Comment (stephenfuqua) [inline Meadowlark-js/backends/meadowlark-search-backend/src/repository/ClientSearch.ts:11]: And using `SEARCH_PROVIDER` as the string here?

Comment (DavidJGapCR) [reply Meadowlark-js/backends/meadowlark-search-backend/src/repository/ClientSearch.ts:11]: Sure, sounds good. Here is where I have some problems. If I uncomment this line, I get an error saying that the variable does not exist.

**Comment by DavidJGapCR** [comment]
Sadly I am finding differences in terms of how opensearch and elasticsearch behave. That is why I worked on a small prove of concept. 
One of the differences I found so far is that one of the test (`/test/integration/QueryOpensearch.test.ts`) **When querying for documents** -> **when querying with parameters** -> **when querying with wrong property**, will behave differently: Elasticsearch will return error: `Unknown column [lastSure], did you mean [lastSurname]?`, while opensearch returns `QUERY_SUCCESS` with zero results (as the existing test shows).

**Comment by DavidJGapCR** [comment]
Other things will have to change, for example:
```
return client.transport.request({
    method: 'POST',
    path: '/opendistro/_doc/_sql',
    body: { query },
  });
```
in `/src/repository/QuerySearch.ts`.

**Pull request closed without merging**